devnet-sdk/system/interfaces.go

```diff
@@ -32,6 +39,8 @@ type Chain interface {
 	GasPrice(ctx context.Context) (*big.Int, error)
 	GasLimit(ctx context.Context, tx TransactionData) (uint64, error)
 	PendingNonceAt(ctx context.Context, address common.Address) (uint64, error)
+	Config() (ChainConfig, error)
+	BlockByNumber(ctx context.Context, number *big.Int) (*coreTypes.Block, error)
```

Comment:
Want to acknowledge for @protolambda that this is probably not ideal. Happy to either block on @pcw109550's https://github.com/ethereum-optimism/optimism/pull/14644, rebase before merging, and then move this to a client interface within Node or merge now and update later.

Reply:
https://github.com/ethereum-optimism/optimism/pull/14644 one is the log refactoring PR. Correct me If I am wrong, but I think you were trying to mention https://github.com/ethereum-optimism/optimism/pull/14660 (Take out RPC functionality as Node interface from Chain Interface).

Reply:
Yes that's correct.